src/context/UserContextProvider.tsx

```diff
@@ -0,0 +1,20 @@
+// context/UserContextProvider.tsx
+
+import React, { ReactNode, useState } from "react";
+
+import { UserContext } from "./userContext";
+import { User } from "../model/UserType";
+
+type UserContextProviderProps = {
+  children: ReactNode;
+};
+
+export const UserContextProvider = ({ children }: UserContextProviderProps) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  return (
+    <UserContext.Provider value={{ user, setUser }}>
```

Comment:
Memoize the value. I have created this for you -> https://stackblitz.com/edit/context-value-memoization?file=src%2FApp.tsx to demonstrate why it is a good idea

Took me an ungodly long time to make the demo. Be grateful

Reply:
thanks, resolved